### PR TITLE
Fix System.Formats.Asn1 CG alert

### PR DESCRIPTION
**Pull request by eerhardt**
## Description

System.Formats.Asn1 `6.0.0` and `8.0.0` have security vulnerabilities. See CVE-2024-38095.

Updating to `8.0.1` to fix these.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5863)

### CODE REVIEW

**Review by sebastienros — APPROVED** [review]

File: Directory.Packages.props [hunk 1]
```diff
@@ -182,18 +182,16 @@
...
     <PackageVersion Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.17.4" />
     <PackageVersion Include="Microsoft.ApplicationInsights.WorkerService" Version="2.22.0" />
-    <!-- Pinned version for Component Governance - Remove when root dependencies are updated -->
+
+    <!-- Pinned versions for Component Governance - Remove when root dependencies are updated -->
     <PackageVersion Include="Azure.Core" Version="1.43.0" />
     <PackageVersion Include="Azure.Identity" Version="1.12.0" />
-    <PackageVersion Include="Newtonsoft.Json" Version="13.0.3" />
     <!-- https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3313 -->
+    <PackageVersion Include="Newtonsoft.Json" Version="13.0.3" />
```
Comment (joperezr) [inline Directory.Packages.props:194]: Is the comment above still accurate?

**Review by joperezr — APPROVED** [review]

File: Directory.Packages.props (hunk 1 above)
Comment (eerhardt) [reply Directory.Packages.props:194]: Yes - that is the reason why we need to "lift" Newtonsoft.Json - because of CosmosDB.

**Pull request merged**